rust/abacus-core/src/types/messages.rs

```diff
@@ -75,30 +70,28 @@ impl Decode for AbacusMessage {
             sender,
             destination: u32::from_be_bytes(destination),
             recipient,
-            nonce: u32::from_be_bytes(nonce),
             body,
         })
     }
 }
 
 impl AbacusMessage {
     /// Convert the message to a leaf
-    pub fn to_leaf(&self) -> H256 {
-        keccak256(self.to_vec()).into()
-    }
-
-    /// Get the encoded destination + nonce
-    pub fn destination_and_nonce(&self) -> u64 {
-        utils::destination_and_nonce(self.destination, self.nonce)
+    pub fn to_leaf(&self, leaf_index: u32) -> H256 {
```

Comment:
ditto here on ensuring we are replicating `abi.encodePacked` correctly- also, if we do stay with this solution, any way we can pull the contents into a function that this `to_leaf` and `RawCommittedMessage`'s `to_leaf` can use?

Reply:
How strongly do you feel about this? IMO, this level of duplication isn't worth the redirection of extraction

Reply:
not too strongly so happy to keep as is